traitsui/qt4/table_editor.py

```diff
@@ -1031,21 +1043,20 @@ def eventFilter(self, obj, event):
 
         vheader = self.verticalHeader()
         if obj is vheader and event.type() == QtCore.QEvent.Type.ContextMenu:
-            event.accept()
+            position = event.globalPos()
             editor = self._editor
             row = vheader.logicalIndexAt(event.pos().y())
             if row == -1:
-                factory = editor.factory
-                if factory.row_factory is not None:
-                    editor.empty_menu.exec_(event.globalPos())
+                empty_menu = editor._create_empty_menu()
+                if empty_menu is not None:
+                    event.accept()
+                    empty_menu.exec_(position)
             else:
                 editor.header_row = row
```

Comment:
Is there other code which is dependent on this mutation of `header_row`? Otherwise I would think it's better to pass `row` to `_create_header_menu`.

Reply:
Yes, unfortunately.  It's used by most of the handler methods for the default menu Actions (`_on_context_insert`, `_on_context_remove`, etc.)